### PR TITLE
[ENG-1611] Department update + mobile view for institutional dashboard

### PR DESCRIPTION
**Pull request by futa-ikeda**
- Ticket: [ENG-1611]
- Feature flag: n/a

## Purpose
- Update mobile view
- Load only 10 depts at a time

## Summary of Changes
- New breakpoint added in CSS to account for mobile view
- Replace use of `loadAll` for departments and just fetch the first 10 depts (these should be the top 10 depts with the most users)
- Update department pie chart to only show the top 10 depts with most users and added an `Other` slice and refactored how that pie chart data is created

## Side Effects
`NA`

## QA Notes
- Please take a look at the Inst. Dashboard in mobile views and make sure nothing is "gross"
- The departments pie chart should have, at most, 11 slices. The top 10 with most users and "Other". Please verify that those numbers are correct when there are less than 10 depts as well as more than 10 depts

[ENG-1611]: https://openscience.atlassian.net/browse/ENG-1611

### CODE REVIEW

File: app/institutions/dashboard/-components/departments-panel/component.ts [hunk 1]
```diff
@@ -22,44 +27,44 @@ export default class DepartmentsPanel extends Component {
         },
     };
 
-    didReceiveAttrs() {
-        if (this.departments) {
-            const departmentNumbers = this.departments.map(x => x.numberOfUsers);
-            this.set('chartHoverIndex', departmentNumbers.indexOf(Math.max(...departmentNumbers)));
-        }
+    @computed('topDepartments')
```
Comment (fabmiz) [inline app/institutions/dashboard/-components/departments-panel/component.ts:30]:
```suggestion
    @computed('topDepartments.[]')
```

File: app/institutions/dashboard/-components/departments-panel/component.ts [hunk 2]
```diff
@@ -22,44 +27,44 @@ export default class DepartmentsPanel extends Component {
...
+    get displayDepartments() {
+        const departments = this.topDepartments.map(({ name, numberOfUsers }) => ({ name, numberOfUsers }));
+        const departmentNumbers = this.topDepartments.map(x => x.numberOfUsers);
+        const otherDepartmentNumber = this.totalUsers - departmentNumbers.reduce((a, b) => a + b);
+
+        return [...departments, { name: this.intl.t('general.other'), numberOfUsers: otherDepartmentNumber }];
     }
 
-    @computed('chartHoverIndex', 'departments')
+    @computed('chartHoverIndex', 'displayDepartments')
```
Comment (fabmiz) [inline app/institutions/dashboard/-components/departments-panel/component.ts:39]:
```suggestion
    @computed('chartHoverIndex', 'displayDepartments.[]', 'chartHoverIndex')
```

File: app/institutions/dashboard/-components/departments-panel/component.ts [hunk 3]
```diff
@@ -22,44 +27,44 @@ export default class DepartmentsPanel extends Component {
...
             datasets: [{
-                data: departmentNumbers,
+                data: displayDepartmentNumbers,
                 backgroundColor: backgroundColors,
             }],
         };
     }
 
-    @computed('chartHoverIndex', 'departments')
+    @computed('chartHoverIndex', 'displayDepartments')
```
Comment (fabmiz) [inline app/institutions/dashboard/-components/departments-panel/component.ts:59]:
```suggestion
    @computed('chartHoverIndex', 'displayDepartments.[]')
```

File: app/institutions/dashboard/-components/departments-panel/component.ts [hunk 4]
```diff
@@ -22,44 +27,44 @@ export default class DepartmentsPanel extends Component {
...
     get activeDepartment(): Department {
-        return this.departments.toArray()[this.chartHoverIndex];
+        return this.displayDepartments[this.chartHoverIndex];
     }
 
-    @computed('activeDepartment', 'departments')
+    @computed('activeDepartment', 'displayDepartments')
```
Comment (fabmiz) [inline app/institutions/dashboard/-components/departments-panel/component.ts:64]:
```suggestion
    @computed('activeDepartment', 'displayDepartments.[]')
```

File: app/institutions/dashboard/route.ts [hunk 5]
```diff
@@ -27,9 +28,19 @@ export default class InstitutionsDashboardRoute extends Route {
                     include: ['summary_metrics'],
                 },
             });
-            const departmentMetrics = yield institution.loadAll('departmentMetrics');
+            const departmentMetrics = yield institution.queryHasMany('departmentMetrics');
             const summaryMetrics = yield institution.summaryMetrics;
-            return { institution, departmentMetrics, summaryMetrics };
+            const userMetricInfo: QueryHasManyResult<never> = yield institution.queryHasMany(
+                'userMetrics',
+                { size: 0 },
```
Comment (fabmiz) [inline app/institutions/dashboard/route.ts:35]: what does `size: 0` do here?

File: app/institutions/dashboard/-components/departments-panel/component.ts [hunk 6]
```diff
@@ -1,13 +1,18 @@
...
 
 export default class DepartmentsPanel extends Component {
-    departments!: InstitutionDepartmentsModel[];
+    @service intl!: Intl;
 
-    chartHoverIndex: number = -1;
+    topDepartments!: InstitutionDepartmentsModel[];
+    totalUsers!: number;
+
+    chartHoverIndex: number = 0;
```
Comment (fabmiz) [inline app/institutions/dashboard/-components/departments-panel/component.ts:15]: with `chartHoverIndex` initialized to 0 as opposed to -1, shouldn't `chartOptions.onHover` callback be updated somehow? Also shouldn't we bind `chartOptions.onHover` to `DepartmentsPanel` `this` context, seeing as we are referencing `this.set` and `this.chartHoverIndex`.

File: app/institutions/dashboard/route.ts (hunk 5 above)
Comment (futa-ikeda) [reply app/institutions/dashboard/route.ts:35]: We use `userMetricsInfo` only to get the total count, so this just makes sure we aren't loading anything extraneous.

**Pull request merged**